File: app/mlboard/api/services/maintenance_log.py
from mlboard.core import models as ms
from mlboard.core import queries as qs
from mlboard.core import db
from fastapi import APIRouter, Query
import typing
from logging import getLogger
from cytoolz.curried import pipe, map
from pydantic import BaseModel
import uuid
import datetime

router = APIRouter()


class MaintenanceLog(BaseModel):
    id: uuid.UUID
    maintenance_id: uuid.UUID
    occurred_date: datetime.datetime


@router.post('/maintenance-log', response_model=uuid.UUID)
async def upsert(payload: MaintenanceLog):
    maintenace_log = ms.MaintenanceLog(**dict(payload))
    async with db.get_conn() as conn:
        queried_id = await qs.MaintenanceLog(conn).upsert(maintenace_log)
    return queried_id


@router.delete('/maintenance-log', response_model=uuid.UUID)
async def delete_by(id: uuid.UUID):
    async with db.get_conn() as conn:
        queried_id = await qs.MaintenanceLog(conn).delete_by(id=id)
    return queried_id


@router.get('/maintenance-log/all')
async def all():
    async with db.get_conn() as conn:
        rows = await qs.MaintenanceLog(conn).all()
    return rows

File: board/src/core/api/maintenanceLog.ts
import { IMaintenanceLog } from "~/core/models"
import axios from 'axios'

interface IResponse {
  id: string
  create_date: string
  maintenance_id: string
  occurred_date: string
}

interface IForm {
  id: string
  maintenanceId: string
  occurredDate: string
}

export const all = async ():Promise<IMaintenanceLog[]> => {
  const res = await axios.get('api/maintenance-log/all')
  const resData = res.data as IResponse[]
  return resData.map(x => { 
    return {
      id: x.id,
      maintenanceId: x.maintenance_id,
      occurredDate: x.occurred_date,
      createDate: x.create_date
    }
  })
}

export const upsert =  async (payload: IForm):Promise<string> => {
  const res = await axios.post('api/maintenance-log', {
    id: payload.id,
    occurred_date: payload.occurredDate,
    maintenance_id: payload.maintenanceId,
  })
  const resData = res.data as string
  return resData
}

export const deleteBy = async (id:string):Promise<string> => {
  const res = await axios.delete('api/maintenance-log', {
    params: {
      id: id
    }
  })
  const resData = res.data as string
  return resData
}
